supportedBackends/htcondor/src/main/scala/cromwell/backend/impl/htcondor/caching/localization/CachedResultLocalization.scala

```diff
@@ -0,0 +1,43 @@
+package cromwell.backend.impl.htcondor.caching.localization
+
+import java.nio.file.{Files, Path, Paths}
+
+import better.files.File
+import cromwell.core.{JobOutput, _}
+import wdl4s.types.{WdlArrayType, WdlFileType}
+import wdl4s.values.{WdlArray, WdlSingleFile, WdlValue}
+
+import scala.util.Try
+
+trait CachedResultLocalization {
+  def localizePathViaSymbolicLink(originalPath: Path, executionPath: Path): Path = {
+    if (File(originalPath).isDirectory) throw new UnsupportedOperationException("Cannot localize directory with symbolic links.")
+    else {
+      File(executionPath).parent.createDirectories()
+      Files.createSymbolicLink(executionPath, originalPath.toAbsolutePath)
+    }
+  }
+
+  def localizeCachedFile(executionPath: Path, output: WdlValue): WdlSingleFile = {
+    val origPath = Paths.get(output.valueString)
+    val newPath = executionPath.toAbsolutePath.resolve(origPath.getFileName)
+    val slPath = localizePathViaSymbolicLink(origPath, newPath)
+    WdlSingleFile(slPath.toString)
+  }
+
+  def localizeCachedOutputs(executionPath: Path, outputs: JobOutputs): Try[JobOutputs] = {
+    Try {
+      outputs.map { output =>
```

Comment:
also no `.` with the curly variant of HOFs (and you'll want to keep the curly variant to be able to `case` 😄 )